src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -834,7 +841,22 @@ public int getSize() {
      *             the io exception
      */
     public PagedIterable<GHUser> listCollaborators() throws IOException {
-        return listUsers("collaborators");
+        return listCollaborators(CollaboratorAffiliation.ALL);
+    }
+
+    /**
+     * Lists up the collaborators on this repository.
+     *
+     * @param affiliation
+     *            Filter users by affiliation
+     * @return Users paged iterable
+     * @throws IOException
+     *             the io exception
+     */
+    public PagedIterable<GHUser> listCollaborators(CollaboratorAffiliation affiliation) throws IOException {
+        Map<String, Object> args = new HashMap<>();
+        args.put("affiliation", affiliation.toString().toLowerCase());
+        return listUsers("collaborators", args);
```

Comment:
Does this work? 
```suggestion
        return listUsers("collaborators", 
            Collections.singletonMap("affiliation", affiliation);
```

Reply:
Take a look at how I've streamlined things now and tell me what you think.